R/01-ckan.R

```diff
@@ -1,196 +1,242 @@
+                         repository = "", network = "", pattern = "", order = TRUE) {
+  res <- callAPI(action = "current_package_list_with_resources", limit = 1000)
+  
+  emptyOut <- data.frame(name = character(),
+                         title = character(),
+                         resources = character())
+  
+  if (!is.null(attr(res, "error"))) {
+    attr(emptyOut, "error") <- attr(res, "error")
+    return(emptyOut)
+  }
+  
+  if (!all(c("name", "title", "resources") %in% names(res))) {
+    return(emptyOut)
+  }
+  
+  res <- res %>%
+    filterNetwork(network = network) %>%
+    filterRepository(repository = repository) %>%
+    filterPattern(pattern = pattern)
     
-    # set selected before sorting !
-    default <- sapply(types, function(type) {
-      match(type, ckanTypes)
-    })
-    default <- default[which(!is.na(default))[1]]
-    selected <- choices[default]
+  resResources <- res[["resources"]]
+  names(resResources) <- res[["name"]]
+  
+  # select not empty files
+  resResources <- lapply(resResources, function(resource) {
+    resource <- resource %>% 
+      filter(.data$format != "", .data$url != "")
+  })
+  
+  resResources <- resResources[sapply(resResources, nrow) > 0]
+  
+  if (length(resResources) == 0) return(emptyOut)
+  # select relevant entries
+  resourcesDF <- lapply(seq_along(resResources), function(i) {
+    # set up df
+    df <- resResources[[i]][c("name", "format", "url")] 
+    df$format <- df$format %>%
+      gsub(pattern = "\\.", replacement = "") %>%
+      tolower()
+    df$repository <- names(resResources[i])
     
-    # sort choices
-    if (sort) {
-      choices <- choices %>% sort()
+    if (length(fileType) > 0) {
+      # filter selected file types
+      df <- df[df$format %in% tolower(fileType), , drop = FALSE]
     }
     
-    # return
-    list(choices = choices,
-         selected = selected)
-  }
-
-getCKANRecordChoices <- function(ckanFiles, sort = TRUE) {
-  if (!is.null(attr(ckanFiles, "error"))) {
-    noChoices <- c("")
-    names(noChoices) <- attr(ckanFiles, "error")
-    return(noChoices)
+    # arrange df
+    df[, c("repository", "name", "format", "url")]
+  }) %>%
+    bind_rows() %>%
+    distinct()
+  
+  if (order) {
+    resourcesDF <- resourcesDF %>%
+      arrange(.data$repository, .data$name)
   }
   
-  choices <- unlist(lapply(ckanFiles, `[[`, "title"))
+  resourcesDF
+}
+
+#' Get File Types
+#' 
+#' @param repository (character) list of relevant file types, e.g. c("xls", "xlsx", "csv", "odt")
+#' @param network (character) name of Pandora network
+#' @param pattern (character) string for meta information search
+#' @param order (logical) if TRUE sort list names alphabetically
+#' 
+#' @return (data.frame) containing available file types within a repository
+#' @export
+getFileTypes <- function(repository = "", network = "", pattern = "", order = TRUE) {
+  res <- callAPI(action = "current_package_list_with_resources", limit = 1000)
```

Comment:
same as above. added description from news.md

```suggestion
#' Get File Types
#'
#' Get all available file types of a repository or those within a 
#' specific network or within a specific repository
#' optional filtering of meta information for a given string
#'
#' @param repository (character) list of relevant file types, e.g. c("xls", "xlsx", "csv", "odt")
#' @param network (character) name of Pandora network
#' @param pattern (character) string for meta information search
#' @param order (logical) if TRUE sort list names alphabetically
#' 
#' @return (data.frame) containing available file types within a repository
#' @export
getFileTypes <- function(repository = "", network = "", pattern = "", order = TRUE) {
  res <- callAPI(action = "current_package_list_with_resources", limit = 1000)
```